bip32/src/derivation_path.rs

```diff
@@ -0,0 +1,33 @@
+//! Derivation paths
+
+use crate::{ChildNumber, Error, Result};
+use alloc::vec::Vec;
+use core::str::FromStr;
+
+/// Derivation paths within a hierarchical keyspace.
+#[derive(Clone, PartialEq, Eq, Debug, Default)]
+pub struct DerivationPath {
+    path: Vec<ChildNumber>,
```

Comment:
nit: dervation paths are limited by spec to at most 256 elements, including the `m`. This type will allow invalid paths